### PR TITLE
reduce false error detection when deploying coredns charm

### PR DESCRIPTION
**Pull request by addyess**
* allow this test to run on a gce cloud if you happen to have one lying around
* better handle transient `errors` in the coredns charm if they go away eventually
* use libjuju to remove coredns charm after it's completed its usefulness

### CODE REVIEW

File: jobs/integration/validation.py [hunk 1]
```diff
@@ -1947,11 +1947,7 @@ async def get_offer():
             coredns = k8s_model.applications["coredns"]
 
         log("Waiting for CoreDNS charm to be ready")
-        while (
-            status := coredns.units[0].workload_status if coredns.units else None
-        ) != "active":
-            assert status not in ("blocked", "error")
-            await asyncio.sleep(5)
+        await k8s_model.wait_for_idle(raise_on_error=False, status="active")
```
Comment (mateoflorido) [inline jobs/integration/validation.py:1950]: I think we should add a comment here so that we can come back later when this issue is addressed.

Comment (addyess) [reply jobs/integration/validation.py:1950]: i'm guessing the offensive (workaroundy) part here is the `raise_on_error=False`

this prevents the method from raising an exception when ANY error occurs.  Now, if the charm isn't active after the standard timeout (10m) then if it's still in error, a timeout error will occur

Comment (mateoflorido) [reply jobs/integration/validation.py:1950]: Alright, that makes sense! :)

**Review by mateoflorido — APPROVED** [review]
LGTM 💯

**Pull request merged**